content/tutorials/history.textile

```diff
@@ -287,6 +374,19 @@ In this demo we use the "history API":/realtime/history to replay up to the last
 
 h2. Download tutorial source code
 
+blang[java].
+  The complete source code for each step of "this tutorial is available on Github":https://github.com/piotrekinventica/tutorials/commits/history-java-rest.
+
+  We recommend that you clone the repo locally:
+
+  bc[sh]. git clone https://github.com/ably/tutorials.git
+
+  Checkout the tutorial branch:
+
+  bc[sh]. git checkout history-java-rest
+
+  And then run the demo on your Android device. Don't forget to replace your @ExampleActivity#API_KEY@ field with "Ably API key":https://support.ably.io/support/solutions/articles/3000030502-setting-up-and-managing-api-keys.
```

Comment:
Some basic instructions on "how to run the demo on your Android device" is needed, even if you link to some other page that provides these instructions.  The point of these tutorials is to allow someone to follow each step and get something working. These instructions assume the user knows how Gradle works and how to run the demo on their device, so unfortunately this is not instructive enough.